src/openid.js

```diff
@@ -38,44 +50,43 @@ const getTokens = (code, state, host) =>
     .then(slackToken => {
       logger.debug('Got token: %s', JSON.stringify(slackToken), {});
 
-      return getUserInfo(slackToken.access_token).then(userInfo => {
-        logger.debug('Got user details: %s', JSON.stringify(userInfo), {});
+      const userInfo = getUserInfoFromTokenResponse(slackToken);
+      logger.debug('Got user details: %s', JSON.stringify(userInfo), {});
 
-        // Slack returns scopes separated by commas
-        // But OAuth wants them to be spaces
-        // https://tools.ietf.org/html/rfc6749#section-5.1
-        // Also, we need to add openid as a scope,
-        // since we stripped it out earlier otherwise Slack would complain
-        const scope = `openid ${slackToken.scope.replace(',', ' ')}`;
+      // Slack returns scopes separated by commas
+      // But OAuth wants them to be spaces
+      // https://tools.ietf.org/html/rfc6749#section-5.1
+      // Also, we need to add openid as a scope,
+      // since we stripped it out earlier otherwise Slack would complain
+      const scope = `openid ${slackToken.scope.replace(',', ' ')}`;
 
-        // ** JWT ID Token required fields **
-        // iss - issuer https url
-        // aud - audience that this token is valid for (SLACK_CLIENT_ID)
-        // sub - subject identifier - must be unique
-        // ** Also required, but provided by jsonwebtoken **
-        // exp - expiry time for the id token (seconds since epoch in UTC)
-        // iat - time that the JWT was issued (seconds since epoch in UTC)
+      // ** JWT ID Token required fields **
+      // iss - issuer https url
+      // aud - audience that this token is valid for (SLACK_CLIENT_ID)
+      // sub - subject identifier - must be unique
+      // ** Also required, but provided by jsonwebtoken **
+      // exp - expiry time for the id token (seconds since epoch in UTC)
+      // iat - time that the JWT was issued (seconds since epoch in UTC)
 
-        return new Promise(resolve => {
-          const payload = {
-            // This was commented because Cognito times out in under a second
-            // and generating the userInfo takes too long.
-            // It means the ID token is empty except for metadata.
-            ...userInfo
-          };
+      return new Promise(resolve => {
+        const payload = {
+          // This was commented because Cognito times out in under a second
+          // and generating the userInfo takes too long.
+          // It means the ID token is empty except for metadata.
+          ...userInfo
+        };
 
-          const idToken = crypto.makeIdToken(payload, host);
-          const tokenResponse = {
-            ...slackToken,
```

Comment:
this would include more data than needed, since user identity comes with the token, and it's already present in the id token